Editorial review: Add docs for scroll snap events (#36057)
mdn/content: files/en-us/web/css/css_scroll_snap/index.md

- [Basic concepts of CSS scroll snap](/en-US/docs/Web/CSS/CSS_scroll_snap/Basic_concepts)
  - : An overview and examples of CSS scroll snap features.

> - [Using scroll snap events](/en-US/docs/Web/CSS/CSS_scroll_snap/Using_scroll_snap_events)

Reviewer: We need to add events and interfaces, (see example at https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_animations#events). Also need to add https://drafts.csswg.org/css-scroll-snap-2/ and a 

```md
> [!NOTE]
> This module also defines the `X` property/pseudoclass, but it is currently not supported in any browser.
```

as needed.

Author: I've added the L2 module, and the events and interfaces section. I've just added `SnapEvent` to the interfaces section. I think it is overkill to separately list `snapTargetBlock` and `snapTargetInline`

Author: OK, added

>   - : A guide to using the scroll snap events — {{domxref("Element/scrollsnapchanging_event", "scrollsnapchanging")}} and {{domxref("Element/scrollsnapchange_event", "scrollsnapchange")}} — to run JavaScript in response to the browser determining that a new snap target is pending or selected.

Reviewer: ```suggestion
  - : A guide to using the {{domxref("Element/scrollsnapchanging_event", "scrollsnapchanging")}} and {{domxref("Element/scrollsnapchange_event", "scrollsnapchange")}} scroll snap events that are fired when the browser determines a new snap target is pending or selected.
```

Author: Yup, like it, updated.
